std/assembly/rt/itcms.ts

```diff
@@ -176,6 +180,7 @@ function step(): usize {
         __visit_members(changetype<usize>(obj) + TOTAL_OVERHEAD, VISIT_SCAN);
       } else {
         visitRoots(VISIT_SCAN);
+        visitStack(VISIT_SCAN);
```

Comment:
According previous `STATE_IDLE` is "fall through" and already called `visitRoots(VISIT_SCAN);`. Could this be simplified to only `visitStack(VISIT_SCAN);` without `visitRoots(VISIT_SCAN)`?

Reply:
It does not typically fall through to here. The common case is that `obj != toSpace`, taking the `then` branch visiting one object and yielding. Only when all the initial roots have been marked it will take the `else` branch, where it re-visits all roots and the stack to not miss a new root added since starting the run, and it repeats that (marking one new root/object and yielding, checking again) until it is guaranteed that no root has been missed. Only then it flips and enters the sweep phase. Not visiting the stack initially is an optimization in that objects on the stack are likely to change significantly, hence only visiting the stack before flipping.